pkgs/tools/misc/gtkclipblock/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, cmake
+, ninja
+, pkg-config
+, gtk2
+, gtk3
+, gtk4
+}:
+
+let
+  distorm-src = fetchFromGitHub {
+    owner = "gdabah";
+    repo = "distorm";
+    rev = "3.5.2b";
+    hash = "sha256-2ftEV3TMS3HT7f96k+Pwt3Mm31fVEXcHpcbbz05jycU=";
+  };
+  source = fetchFromGitHub {
+    owner = "notpeelz";
+    repo = "gtkclipblock";
+    rev = "v0.2.2";
+    hash = "sha256-ok/D7M0KekN8zf8AzhcOLtedbYVRHHv3m9zEHsJfcPM=";
+    fetchSubmodules = true;
+  };
+in
+stdenv.mkDerivation {
+  pname = "gtkclipblock";
+  version = "0.2.2";
+
+  src = source;
+
+  nativeBuildInputs = [ meson cmake ninja pkg-config ];
+
+  buildInputs = [ gtk2 gtk3 gtk4 ];
```

Comment:
Just to know.
Is GTK2 an optional dependency?
I ask this because GTK 2 is deprecated.

Reply:
Lemme see if I understood.

The options below are mutually exclusive? Only one can be checked at a time?

- [ ] gtk2
- [ ] gtk3
- [ ] gtk4
- [ ] gtk2 and gtk3 and gtk4

Reply:
That's the options gtkclipblock gives, yes. The only way to support gtk3 and gtk4 without gtk2 is to edit `meson.build` and comment out a part of it. Which is rather hacky, but can be done if we'd rather not support gtk2 here.

Reply:
Here's the derivation that does this, is this alright or is it too hacky? 

https://gist.github.com/UARTman/a39e8f7931019044f5005127ab6bbb7b

Reply:
Since it requires editing the meson file to such a huge extent, it is better to suggest this change to the upstream project instead of keeping the hack in Nixpkgs.

For now let's bundle everything.

Reply:
Looking at the patch, can't we just edit the if statement to never execute?